src/region.js

```diff
@@ -156,31 +156,19 @@ const Region = MarionetteObject.extend({
     // always restore the el to ensure the regions el is present before replacing
     this._restoreEl();
 
-    const parent = this.el.parentNode;
-
-    parent.replaceChild(view.el, this.el);
+    this.$el.replaceWith(view.el);
     this._isReplaced = true;
   },
 
   // Restore the region's element in the DOM.
   _restoreEl() {
-    if (!this.currentView) {
-      return;
-    }
-
     const view = this.currentView;
-    const parent = view.el.parentNode;
 
-    if (!parent) {
-      return;
+    if (view && view.el && view.el.parentNode) {
+      // Replace using library to ensure proper cleanup.
```

Comment:
> any events added to the regions el before the show would be lost in this case.

Trying to think of a test for this scenario.

In the mean time I can revert the conditions.

> In the very least it'd probably be good to abstract this.$el.replaceWith to a private function such that one function could be overridden

I agree. The code base should definitely being going that route. I'll look to see if there's an issue to create some sort of mixin that abstract DOM interaction. Otherwise, I'll outline a something later.